rootsanalyzer/rootset_id_test.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rootsanalyzer
+
+import (
+	"encoding/hex"
+	"testing"
+
+	"github.com/google/certificate-transparency-go/x509"
+	"github.com/google/certificate-transparency-go/x509util"
+	"github.com/google/monologue/storage"
+	"github.com/google/monologue/testdata"
+)
+
+func TestCertIDOnNil(t *testing.T) {
+	var m SHAIDMaker
+	_, gotErr := m.GenerateCertID(nil)
+	if gotErr == nil {
+		t.Errorf("Error expected, got nil")
+	}
+}
+
+func TestCertIDOnValid(t *testing.T) {
+	decoded, _ := hex.DecodeString("86d8219c7e2b6009e37eb14356268489b81379e076e8f372e3dde8c162a34134")
```

Comment:
nit: Shouldn't really ignore this error. While the test would fail if the hex string was invalid, it'd be in a way that didn't make it obvious what the problem was. You could call `hexDecodeValid` here instead?

Reply:
Done